### ISSUE TITLE
[GIP] dropping ogc-server-statistics and analytics

### ISSUE THREAD

**Issue by fvanderbiest** [georchestra/improvement-proposals#5]
### Who ?

Camptocamp, with funding from MEL

### Target Module

The `ogc-server-statistics` logger will be removed.
Which implies also a removal of the analytics webapp and a rework of the front console application.

### What ?

As said above, we plan to remove the OGC logging feature from geOrchestra core.

### Why ?

We're preparing the replacement of the older security-proxy by the georchestra-gateway.

1. the georchestra-gateway is not able to host a log4j1 logger like `ogc-server-statistics`
2. storing ogc access logs in a postgresql database was not a good idea, since it quickly became "big data"

### How ?

Essentially `git rm -rf analytics` and `git rm -rf ogc-server-statistics`.

### Any potential pitfalls and ways to circumvent them ?

There's no plan yet to provide an equivalent feature by <<insert here any fancy tech like ELK or ... >>
Maybe we should ?

### When ?

One should expect geOrchestra 24.0 to be free from ogc-server-statistics, which means **funding will be required to get the equivalent feature** by then.

### State of the vote:

| PSC members              | vote |
|-------------------------:|------|
| Fabrice Phung            |      |
| François Van Der Biest   |   +1   |
| Pierre Mauduit           |      |
| Landry Breuil            |   +1   |
| Stéphane Mével-Viannay   |      |
| Maël Reboux              |   +1   |
| Pierre Jégo              |      |
| Jean Pommier             |   +1   |
| Catherine Piton-Morales  |      |

**Comment by jeanpommier** [comment]
I agree on removing it, but we should consider a replacement solution. Globally, I have the feeling we should ease or at least document the integration of analytics tools, since it is a quite common need

Maybe this could be an interesting workshop during the geocom/following codesprint

**Comment by fvanderbiest** [comment]
> Concerning this point, what about use : https://github.com/timescale/timescaledb ?

Love it !
Thanks for the hint.

**Comment by landryb** [comment]
fwiw i use influxdb for similar needs, but they're on the same level with timescaledb.

**Comment by jusabatier** [comment]
> Can you remind us how you use it / feed the logs into it ?

Here is some example config to feed database via log4j2 using JDBC appenders (commented) : https://github.com/georchestra/cadastrapp/blob/master/cadastrapp/src/main/resources/log4j2.properties

It's feed same way as postgresql as it's an extension.

And you can find how to configure retention in the timescaledb docs : https://docs.timescale.com/timescaledb/latest/how-to-guides/data-retention/

**Comment by pierrejego** [comment]
```
Any potential pitfalls and ways to circumvent them ?
There's no plan yet to provide an equivalent feature by <<insert here any fancy tech like ELK or ... >>
Maybe we should ?
```
as @jusabatier @landryb and @jeanpommier 
I think ogcstatistics should not be fully removed until some replacement is found.
For example, we could remove it from the console, but let the possibility to install ogcstatistic apps when using security proxy.

Another idea, without changing architecture or add more framework and since elasticsearch and kibana are already installed, we could probably test some simple logs insertion via logstash and create a route to get Kibana accessible for admin user. With a specific dashboard.

```
Why ?
We're preparing the replacement of the older security-proxy by the georchestra-gateway.
```
This point will probably need more explantation, I know we already spoke about it, but a specific point should be done on this important point when it will occur. Have you an idea when you'd like to replace security-proxy by georchestra-gateway ?

**Comment by pierrejego** [comment]
> From what I could see ES is a sinkhole for resources

True, but already installed for Geonetwork4 so could be shared

**Comment by landryb** [comment]
> > From what I could see ES is a sinkhole for resources
> 
> True, but already installed for Geonetwork4 so could be shared

that's more or less discouraged, as kibana is configured for GN indexes only, and there's some hairy url rewriting being done too...

**Comment by jeanpommier** [comment]
And the default usage made by GN4 (index metadata only) is quite moderate, which allows a "relatively light" ES setup. 

Logs are known to become quickly massive data, specially if we want some retention time, which we will need for analytics. 

I'd rather have some experiments first with lighter tools like loki. How far did you go with Loki, @landryb  ?

**Comment by bchartier** [comment]
I apologize in advance for inserting noise in this discussion:
- any update from the discussions on this topic that occured yesterday during geOcom 2023?
- any thought about InfluxDb for this specific use case?

**Comment by fvanderbiest** [comment]
Florent Berault from MEL says that his needs for analytics is more than just OGC WMS/WFS/etc.
Ideally the Data API should also be part of it. 
Since it is "OGC API Features"-based, it makes a lot of sense to me.

**Comment by jeanpommier** [comment]
Hi @fvanderbiest 
Could you be more specific about what it would imply ? What would be expected ?

**Comment by MaelREBOUX** [comment]
Depends on the vote for #11

**Comment by fvanderbiest** [comment]
Proposal #11 adopted, let's vote for this one.

**Comment by jeanpommier** [comment]
+1

**Comment by MaelREBOUX** [comment]
+1

**Comment by fphg** [comment]
+1

**Comment by pmauduit** [comment]
+1

**Comment by catmorales** [comment]
+1

**Comment by smevel** [comment]
+1

**Comment by pierrejego** [comment]
+1

**Comment by MaelREBOUX** [comment]
Thank you PSC members.

This GIP is adopted.
@fvanderbiest @pmauduit @f-necas Please report here development advances.
